From 02d64f4c6b87362a88013b5f2547ae9070e22d99 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 15 May 2026 16:30:40 +0800
Subject: [PATCH 518/521] AOSCOS: drm: cix: drop duplicate variable declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Whilst building with CONFIG_WERROR enabled, the following error occurs:

  drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c: In function ‘linlondp_split_sbs_master_data_flow’:
  drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c:1246:37: error: variable ‘slave_valid_w’ set but not used [-Werror=unused-but-set-variable]
   1246 |                 u32 master_valid_w, slave_valid_w;
        |
  drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c:1246:21: error: variable ‘master_valid_w’ set but not used [-Werror=unused-but-set-variable]
   1246 |                 u32 master_valid_w, slave_valid_w;
        |

Remove the duplicate variable declaration to fix build.

Fixes: "CIXTECH: drm: add cix linlon dp driver"
Link: https://github.com/cixtech/cix-linux-main/issues/26
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c b/drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c
index e0e3547116d69..06089787cd338 100644
--- a/drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c
+++ b/drivers/gpu/drm/cix/linlon-dp/linlondp_pipeline_state.c
@@ -1243,7 +1243,6 @@ linlondp_split_sbs_master_data_flow(struct linlondp_crtc_state *kcrtc_st,
 		u32 src_y = master->in_y;
 		u32 src_w = master->in_w;
 		u32 src_h = master->in_h;
-		u32 master_valid_w, slave_valid_w;
 		u32 h_scaling_ratio = 0;
 
 #ifdef SBS_NO_OVERLAP_SCALING
@@ -1275,8 +1274,8 @@ linlondp_split_sbs_master_data_flow(struct linlondp_crtc_state *kcrtc_st,
 		/* When overlap is needed, ensure we don't fetch outside of the
 		 * layer when adding the overlap
 		 */
-		master_valid_w = master->out_w;
-		slave_valid_w = slave->out_w;
+		u32 master_valid_w = master->out_w;
+		u32 slave_valid_w = slave->out_w;
 
 #ifdef SBS_NO_OVERLAP_SCALING
 		master->out_w += min(kcrtc->sbs_overlap, slave_valid_w);
-- 
2.52.0

